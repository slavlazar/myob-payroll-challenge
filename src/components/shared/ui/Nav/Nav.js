import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import {
  NavContainer,
  LogoContainer,
  Inner,
  LinksContainer,
  SkipNavigationLink,
} from './Nav.style';
import Logo from '../Logo/Logo';
import Container from '../Container';

/**
 * Renders the MYOB logo
 */
const generateLogo = () => (
  <LogoContainer>
    <Link to="/">
      <Logo />
    </Link>
  </LogoContainer>
);

/**
 * Generates nav links with React Router
 * @param {array} links Renders React Router Dom links
 */
const generateNavLinks = links => (
  <LinksContainer className="nav-links-container">
    {links.map(link => (
      <li key={link.label}>
        <Link to={link.url}>{link.label}</Link>
      </li>
    ))}
  </LinksContainer>
);

/**
 * Renders a nav with position set to top or bottom based on
 * props
 * @param {object} Nav props
 */
const Nav = ({ showLogo, links, position }) => (
  <NavContainer navPosition={position}>
    <Container>
      <Inner showLogo={showLogo} navPosition={position}>
        {showLogo && generateLogo()}
        {generateNavLinks(links)}
      </Inner>
      {position === 'top' && (
        <SkipNavigationLink id="header-nav-skip-navigation" href="#main">
          Skip to content
        </SkipNavigationLink>
      )}
    </Container>
  </NavContainer>
);

Nav.propTypes = {
  links: PropTypes.arrayOf(
    PropTypes.shape({
      url: PropTypes.string,
      label: PropTypes.string,
    })
  ),
  showLogo: PropTypes.bool,
  position: PropTypes.oneOf(['top', 'bottom']),
};

Nav.defaultProps = {
  links: [],
  showLogo: true,
  position: 'top',
};

export default Nav;
