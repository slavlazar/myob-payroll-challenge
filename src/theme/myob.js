const fontFamilyMYOBRegular = 'NeutroMYOB-Regular';
const fontFamilyMYOBMedium = 'NeutroMYOB-Medium';
const fontFamilyMYOBStrong = 'NeutroMYOB-Demi';

export default {
  colour: {
    primary: '#8241AA',
    secondary: '#ed005f',
    text: '#515e6d',
    heading: '#2e3e4f',
    background: '#f3f4f5',
    footerBackground: '#1b2830',
    navBackground: '#ffffff',
    buttonPrimary: '#ffffff',
    border: '#efefef',
    buttonPrimaryBackground: '#00aa65',
    alert: {
      default: '#f5f5f5',
      success: '#00d1b2',
      danger: '#fff5f7',
    },
  },
  font: {
    size: {
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      large: '1.8rem',
      xlarge: '2.2rem',
    },
    variants: {
      body: {
        family: fontFamilyMYOBRegular,
        style: 'normal',
        weight: 400,
        src: [
          'url(https://assets.digital.myob.com/fonts/NeutroMYOB-Regular.woff2)',
          'url(https://assets.digital.myob.com/fonts/NeutroMYOB-Regular.woff)',
        ],
      },
      heading: {
        family: fontFamilyMYOBMedium,
        style: 'normal',
        weight: 600,
        src: [
          'url(https://assets.digital.myob.com/fonts/NeutroMYOB-Medium.woff2)',
          'url(https://assets.digital.myob.com/fonts/NeutroMYOB-Medium.woff)',
        ],
      },
      strong: {
        family: fontFamilyMYOBStrong,
        style: 'normal',
        weight: 700,
        src: [
          'url(https://assets.digital.myob.com/fonts/NeutroMYOB-Demi.woff2)',
          'url(https://assets.digital.myob.com/fonts/NeutroMYOB-Demi.woff)',
        ],
      },
    },
    family: {
      body: fontFamilyMYOBRegular,
      heading: fontFamilyMYOBMedium,
      strong: fontFamilyMYOBStrong,
      fallback:
        '-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Arial,sans-serif',
    },
    weight: {
      bold: 600,
      normal: 400,
    },
  },
  lineHeight: {
    text: 1.7,
    heading: 1.5,
  },
  media: {
    large: 992, // mobile-first
    desktop: 768, // mobile-first
    tablet: 576, // mobile-first
  },
};
