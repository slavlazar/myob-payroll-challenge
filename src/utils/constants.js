export const LOGIN_URL = '/login';
export const API_LOGIN_ENDPOINT = '';
