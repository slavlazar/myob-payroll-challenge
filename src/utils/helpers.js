/**
 * Helper method to get file extension of given file name
 * @param {string} filename Full filename
 * @return {string} Given filename, will return the extension
 */
export const getFileExtension = filename =>
  filename
    .split('.')
    .pop()
    .replace(/[\W_]+/g, '');

/**
 * Helper method to general initial values from a given schema
 * @param {array} fields Array of fields to generate initial values
 * @return {object} Object of generated initial values from schema
 */
export const generateInitialValuesFromSchema = (fields = []) =>
  fields.reduce((accum, key) => ({ ...accum, [key]: '' }), {});

/**
 * Helper method to determine if given object is empty
 * @param {object} obj Object to check if empty
 * @return {boolean} Given object, return true if empty, false otherwise
 */
export const isObjectEmpty = obj =>
  Object.keys(obj).length === 0 && obj.constructor === Object;

/**
 * Helper method to log error messages to console
 * Curried function that first takes name of group,
 * followed by console messages
 * @param {string} groupName Name of group to log
 * @returns {[function, void]} Either returns a curried function or logs
 * to console
 */
export const logError = groupName => (...consoleMsg) => {
  console.group(groupName);
  consoleMsg.forEach(msg => {
    console.error(msg);
  });
  console.groupEnd();
};

/**
 * Helper method to round number to nearest integer
 * @param {number} val Number to round
 * @return {number} Rounded number
 */
export const round = val => Math.round(val);

/**
 *
 * @param {number} val Number to format
 * @param {boolean} roundValue Whether to round the value before formatting
 * @return {string} Formatted currency value
 */
export const formatCurrency = (val, roundValue = true) => {
  const valToConvert = roundValue ? round(val) : val;

  return Number(valToConvert).toLocaleString('en-AU', {
    style: 'currency',
    currency: 'AUD',
    minimumFractionDigits: 2,
  });
};
