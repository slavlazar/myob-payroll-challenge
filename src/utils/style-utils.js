import React from 'react';
import { mount } from 'enzyme';
import { css, ThemeProvider } from 'styled-components';

import theme from '../theme';

const { media: sizes } = theme;

// Media query, usage: ${media.tablet`padding: 1.5rem`}
// min-width used for mobile-first design
export const media = Object.keys(sizes).reduce((acc, label) => {
  acc[label] = (...args) => css`
    @media (min-width: ${sizes[label] / 16}em) {
      ${css(...args)}
    }
  `;

  return acc;
}, {});

export const renderWithTheme = component => {
  return mount(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
};
