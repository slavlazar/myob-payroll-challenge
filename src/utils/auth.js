import axios from 'axios';

import { API_LOGIN_ENDPOINT } from './constants';

/**
 * Login method that calls upon our API endpoint to log the user in
 * @param {object} Object of email and password keys to log user in with
 * @async
 */
const login = async ({ email, password }) => {
  const { data: userInfo } = await axios.post(API_LOGIN_ENDPOINT, {
    email,
    password,
  });

  return userInfo;
};

export default {
  login,
};
