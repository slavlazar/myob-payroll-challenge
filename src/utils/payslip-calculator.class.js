import moment from 'moment';

import * as TAXABLE_INCOME from '../config/tax-rates.json';

import { logError as logErrorGroup, formatCurrency } from './helpers';

/** Class representing Payslip calculator. */
class PayslipCalculator {
  static logError = logErrorGroup('PayslipCalculator');

  /**
   * Create a payslip calculator.
   * @param {string} firstName - Employee first name.
   * @param {string} lastName - Employee last name.
   * @param {number} annualSalary - Employee annual salary.
   * @param {number} superRate - Employee super rate.
   */
  constructor(employeeDetails) {
    ({
      firstName: this._firstName,
      lastName: this._lastName,
      annualSalary: this._annualSalary,
      superRate: this._superRate,
    } = employeeDetails);

    this._grossIncome = 0;
    this._incomeTax = 0;
    this._netIncome = 0;
    this._superTotal = 0;
    this._payFrequency = 'Monthly';

    this._determinePayPeriod(employeeDetails.payPeriod);
  }

  /**
   * Get employee full name.
   * @return {string} Employee first name and last name.
   */
  get name() {
    return `${this._firstName} ${this._lastName}`;
  }

  /**
   * Get employee pay period.
   * @return {string} Employee pay period based on month and year.
   */
  get payPeriod() {
    return this._payPeriod;
  }

  /**
   * Get employee pay frequency.
   * @return {string} Employee pay frequency - Defaults to monthly for demo.
   */
  get payFrequency() {
    return this._payFrequency;
  }

  /**
   * Get employee annual salary.
   * @return {string} Employee salary formatted to AUD currency.
   */
  get annualSalary() {
    return formatCurrency(this._annualSalary);
  }

  /**
   * Get employee gross income.
   * @return {string} Employee gross income formatted to AUD currency.
   */
  get grossIncome() {
    return formatCurrency(this._grossIncome);
  }

  /**
   * Get employee income tax.
   * @return {string} Employee income tax formatted to AUD currency.
   */
  get incomeTax() {
    return formatCurrency(this._incomeTax);
  }

  /**
   * Get employee net income.
   * @return {string} Employee net income formatted to AUD currency.
   */
  get netIncome() {
    return formatCurrency(this._netIncome);
  }

  /**
   * Get employee super total.
   * @return {string} Employee super total formatted to AUD currency.
   */
  get superTotal() {
    const superCalc = this._superRate / 100;
    return formatCurrency(this._grossIncome * superCalc);
  }

  /**
   * Get employee pay.
   * @return {string} Employee pay for the month, formatted to AUD currency.
   */
  get pay() {
    return formatCurrency(this._netIncome - this._superTotal);
  }

  /**
   * Get total tax amount for employee, given taxable amount,
   * tax rate and base tax amount.
   * @static
   * @param {number} Taxable amount of employee
   * @returns [{function}, {function}, {number}] Total tax amount for employee
   */
  static __totalTaxAmount(taxable) {
    return taxRate => baseTaxAmount => baseTaxAmount + taxable * taxRate;
  }

  /**
   * Determine pay period (currently monthly) of given date
   * @param {date} payPeriod Date object of current pay period
   * @return {string} Formatted string of pay period from start of
   * month to end of month
   */
  _determinePayPeriod(payPeriod) {
    const payPeriodMoment = moment(payPeriod);
    if (payPeriodMoment.isValid()) {
      const dateFormat = 'DD MMMM';
      const endPayPeriodMoment = payPeriodMoment.clone();
      payPeriodMoment.startOf('month');
      endPayPeriodMoment.endOf('month');

      this._payPeriod = `${payPeriodMoment.format(
        dateFormat
      )} to ${endPayPeriodMoment.format(dateFormat)}`;
    }
  }

  /**
   * Calculate gross based on annual salary / 12 months
   * @return {number} Gross income calculated
   */
  _calculateGross() {
    let grossIncome = 0;

    try {
      grossIncome = this._annualSalary / 12;
    } catch (e) {
      PayslipCalculator.logError(
        '_calculateGross',
        `Salary: ${this._annualSalary}`,
        e
      );
    }

    return grossIncome;
  }

  /**
   * Calculate income tax based on tax limit
   * Method will first reverse the provided TAXABLE_INCOME
   * values (so we can traverse in reverse order), then map over
   * this reversed array to find the correct taxable income based
   * on the employees annual salary
   * @return {number} Income tax based on employee salary and taxable
   * income rules
   */
  _calculateIncomeTax() {
    let totalTax = 0;

    // Avoid mutation/side effect
    const reversedTaxRates = [...TAXABLE_INCOME.default].reverse();

    reversedTaxRates.some(tax => {
      if (this._annualSalary > tax.limit) {
        try {
          const taxable = PayslipCalculator.__totalTaxAmount(
            this._annualSalary - tax.limit
          );
          totalTax = taxable(tax.taxRate)(tax.baseTaxAmount);
        } catch (e) {
          PayslipCalculator.logError(
            'Could not calculate income tax',
            `this._annualSalary: ${this._annualSalary}`,
            `tax object: ${tax}`,
            e
          );
        }
        return true;
      }

      return false;
    });

    return totalTax > 0 ? totalTax / 12 : 0;
  }

  /**
   * Calculates net income based on gross income and income tax
   * @return {number} Net income total
   */
  _calculateNetIncome() {
    return this._grossIncome - this._incomeTax;
  }

  /**
   * Calculates Super based on gross income and super rate
   * @return {number} Super value
   */
  _calculateSuperTotal() {
    const superCalc = this._superRate / 100;
    return this._grossIncome * superCalc;
  }

  /**
   * Public method to generate payslip based on provided variables
   * @return {void}
   * @public
   */
  generate() {
    this._grossIncome = this._calculateGross();
    this._incomeTax = this._calculateIncomeTax();

    this._netIncome = this._calculateNetIncome();
    this._superTotal = this._calculateSuperTotal();
  }
}

export default PayslipCalculator;
