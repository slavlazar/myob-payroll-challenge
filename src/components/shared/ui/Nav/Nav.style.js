import styled from 'styled-components';

import { media } from '../../../../utils/style-utils';

export const Inner = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: ${props => (props.showLogo ? 'space-between' : 'flex-end')};
  padding: ${props =>
    props.navPosition === 'bottom' ? '1.4rem 0' : '0.9rem 0'};
`;

export const LogoContainer = styled.div`
  width: ${props => props.logoWidth || '70px'};
  display: block;
`;

export const LinksContainer = styled.ul`
  display: flex;
  flex-direction: column;
  list-style: none;
  margin: 0;
  align-items: center;

  li {
    display: inline-block;
    text-align: left;

    &:not(:first-child) {
      margin-top: 0.8rem;
    }

    a {
      &,
      &:link,
      &:visited {
        display: inline-block;
        text-decoration: none;
        font-family: inherit;
        padding: 0.8rem 1.2rem;
        border-radius: 0.4rem;
        color: ${props => props.theme.colour.text};
        font-size: ${props => props.theme.font.size.small};
        transition: all 0.2s;
      }

      &:hover,
      &:focus {
        background-color: ${props => props.theme.colour.primary};
        color: #ffffff;
      }
    }
  }

  ${media.tablet`
    flex-direction: row;
    
    li {
      &:not(:first-child) {
        margin-top: 0;
        margin-left: 10rem;
      }
    }
  `};
`;

export const SkipNavigationLink = styled.a`
  position: absolute;
  top: -1000px;
  left: -1000px;
  height: 1px;
  width: 1px;
  text-align: left;
  overflow: hidden;

  &:active,
  &:focus,
  &:hover {
    left: 0;
    top: 0;
    width: auto;
    height: auto;
    overflow: visible;
  }
`;

export const NavContainer = styled.nav`
  background-color: #ffffff;
  position: relative;
  border-bottom: ${props => `1px solid ${props.theme.colour.border}`};

  ${({ navPosition, theme }) =>
    navPosition === 'bottom' &&
    `
    border-bottom: none;
    border-top: 1px solid ${theme.colour.border}; 
  `}
`;
