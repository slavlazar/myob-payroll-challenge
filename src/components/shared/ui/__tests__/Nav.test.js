import React from 'react';
import { shallow } from 'enzyme';
import { Link } from 'react-router-dom';

import Nav from '../Nav/Nav';
import Logo from '../Logo/Logo';
import { NavContainer, Inner } from '../Nav/Nav.style';
import { renderWithTheme } from '../../../../utils/style-utils';

import theme from '../../../../theme';

describe('Components: Shared: UI: Nav', () => {
  describe('Nav', () => {
    it('renders correctly', () => {
      const wrapped = shallow(<Nav />);
      expect(wrapped).toMatchSnapshot();
    });

    it('should render a logo', () => {
      const wrapped = shallow(<Nav showLogo={false} />);
      expect(wrapped.find(Logo)).not.toHaveLength(1);
      wrapped.setProps({ showLogo: true });
      expect(wrapped.find(Logo)).toHaveLength(1);
    });

    it('should generate a link', () => {
      const links = [
        {
          url: '/test',
          label: 'Test',
        },
      ];

      const wrapped = shallow(<Nav links={links} />);
      const linksContainer = wrapped.find('.nav-links-container');
      const link = linksContainer.find(Link);
      expect(link).toHaveLength(1);
      expect(link.prop('to')).toEqual('/test');
    });

    it('should render a skip navigation link for the top nav', () => {
      const wrapped = shallow(<Nav position="bottom" />);
      expect(wrapped.find('#header-nav-skip-navigation')).not.toHaveLength(1);
      wrapped.setProps({ position: 'top' });
      expect(wrapped.find('#header-nav-skip-navigation')).toHaveLength(1);
    });
  });

  describe('NavStyle', () => {
    it('renders NavContainer correctly', () => {
      const wrapped = renderWithTheme(<NavContainer />);
      expect(wrapped).toMatchSnapshot();
    });

    it('renders Inner correctly', () => {
      const wrapped = renderWithTheme(<Inner />);
      expect(wrapped).toMatchSnapshot();
    });

    it('should render a bottom styled container', () => {
      const wrapped = renderWithTheme(<NavContainer navPosition="bottom" />);
      expect(wrapped).toHaveStyleRule('border-bottom', 'none');
      expect(wrapped).toHaveStyleRule(
        'border-top',
        `1px solid ${theme.colour.border}`
      );
    });

    it('should justify content accordingly with no logo', () => {
      const wrapped = renderWithTheme(<Inner showLogo={false} />);
      expect(wrapped).not.toHaveStyleRule('justify-content', 'space-between');
      expect(wrapped).toHaveStyleRule('justify-content', 'flex-end');
    });
  });
});
