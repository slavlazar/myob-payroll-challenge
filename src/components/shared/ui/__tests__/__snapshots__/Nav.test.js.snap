// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Shared: UI: Nav Nav renders correctly 1`] = `
<styled.nav
  navPosition="top"
>
  <styled.div>
    <styled.div
      navPosition="top"
      showLogo={true}
    >
      <styled.div>
        <Link
          to="/"
        >
          <Logo
            alt="MYOB"
            className=""
          />
        </Link>
      </styled.div>
      <styled.ul
        className="nav-links-container"
      />
    </styled.div>
    <styled.a
      href="#main"
      id="header-nav-skip-navigation"
    >
      Skip to content
    </styled.a>
  </styled.div>
</styled.nav>
`;

exports[`Components: Shared: UI: Nav NavStyle renders Inner correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 0.9rem 0;
}

<ThemeProvider
  theme={
    Object {
      "colour": Object {
        "alert": Object {
          "danger": "#fff5f7",
          "default": "#f5f5f5",
          "success": "#00d1b2",
        },
        "background": "#f3f4f5",
        "border": "#efefef",
        "buttonPrimary": "#ffffff",
        "buttonPrimaryBackground": "#00aa65",
        "footerBackground": "#1b2830",
        "heading": "#2e3e4f",
        "navBackground": "#ffffff",
        "primary": "#8241AA",
        "secondary": "#ed005f",
        "text": "#515e6d",
      },
      "font": Object {
        "family": Object {
          "body": "NeutroMYOB-Regular",
          "fallback": "-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Arial,sans-serif",
          "heading": "NeutroMYOB-Medium",
          "strong": "NeutroMYOB-Demi",
        },
        "size": Object {
          "large": "1.8rem",
          "medium": "1.6rem",
          "small": "1.4rem",
          "xlarge": "2.2rem",
          "xsmall": "1.2rem",
        },
        "variants": Object {
          "body": Object {
            "family": "NeutroMYOB-Regular",
            "src": Array [
              "url(https://assets.digital.myob.com/fonts/NeutroMYOB-Regular.woff2)",
              "url(https://assets.digital.myob.com/fonts/NeutroMYOB-Regular.woff)",
            ],
            "style": "normal",
            "weight": 400,
          },
          "heading": Object {
            "family": "NeutroMYOB-Medium",
            "src": Array [
              "url(https://assets.digital.myob.com/fonts/NeutroMYOB-Medium.woff2)",
              "url(https://assets.digital.myob.com/fonts/NeutroMYOB-Medium.woff)",
            ],
            "style": "normal",
            "weight": 600,
          },
          "strong": Object {
            "family": "NeutroMYOB-Demi",
            "src": Array [
              "url(https://assets.digital.myob.com/fonts/NeutroMYOB-Demi.woff2)",
              "url(https://assets.digital.myob.com/fonts/NeutroMYOB-Demi.woff)",
            ],
            "style": "normal",
            "weight": 700,
          },
        },
        "weight": Object {
          "bold": 600,
          "normal": 400,
        },
      },
      "lineHeight": Object {
        "heading": 1.5,
        "text": 1.7,
      },
      "media": Object {
        "desktop": 768,
        "large": 992,
        "tablet": 576,
      },
    }
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: ",
              [Function],
              ";
  padding: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      />
    </StyledComponent>
  </styled.div>
</ThemeProvider>
`;

exports[`Components: Shared: UI: Nav NavStyle renders NavContainer correctly 1`] = `
.c0 {
  background-color: #ffffff;
  position: relative;
  border-bottom: 1px solid #efefef;
}

<ThemeProvider
  theme={
    Object {
      "colour": Object {
        "alert": Object {
          "danger": "#fff5f7",
          "default": "#f5f5f5",
          "success": "#00d1b2",
        },
        "background": "#f3f4f5",
        "border": "#efefef",
        "buttonPrimary": "#ffffff",
        "buttonPrimaryBackground": "#00aa65",
        "footerBackground": "#1b2830",
        "heading": "#2e3e4f",
        "navBackground": "#ffffff",
        "primary": "#8241AA",
        "secondary": "#ed005f",
        "text": "#515e6d",
      },
      "font": Object {
        "family": Object {
          "body": "NeutroMYOB-Regular",
          "fallback": "-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Arial,sans-serif",
          "heading": "NeutroMYOB-Medium",
          "strong": "NeutroMYOB-Demi",
        },
        "size": Object {
          "large": "1.8rem",
          "medium": "1.6rem",
          "small": "1.4rem",
          "xlarge": "2.2rem",
          "xsmall": "1.2rem",
        },
        "variants": Object {
          "body": Object {
            "family": "NeutroMYOB-Regular",
            "src": Array [
              "url(https://assets.digital.myob.com/fonts/NeutroMYOB-Regular.woff2)",
              "url(https://assets.digital.myob.com/fonts/NeutroMYOB-Regular.woff)",
            ],
            "style": "normal",
            "weight": 400,
          },
          "heading": Object {
            "family": "NeutroMYOB-Medium",
            "src": Array [
              "url(https://assets.digital.myob.com/fonts/NeutroMYOB-Medium.woff2)",
              "url(https://assets.digital.myob.com/fonts/NeutroMYOB-Medium.woff)",
            ],
            "style": "normal",
            "weight": 600,
          },
          "strong": Object {
            "family": "NeutroMYOB-Demi",
            "src": Array [
              "url(https://assets.digital.myob.com/fonts/NeutroMYOB-Demi.woff2)",
              "url(https://assets.digital.myob.com/fonts/NeutroMYOB-Demi.woff)",
            ],
            "style": "normal",
            "weight": 700,
          },
        },
        "weight": Object {
          "bold": 600,
          "normal": 400,
        },
      },
      "lineHeight": Object {
        "heading": 1.5,
        "text": 1.7,
      },
      "media": Object {
        "desktop": 768,
        "large": 992,
        "tablet": 576,
      },
    }
  }
>
  <styled.nav>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  background-color: #ffffff;
  position: relative;
  border-bottom: ",
              [Function],
              ";

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.nav",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "nav",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <nav
        className="c0"
      />
    </StyledComponent>
  </styled.nav>
</ThemeProvider>
`;
